packages/mui-lab/src/Masonry/Masonry.js

```diff
@@ -211,73 +212,83 @@ const Masonry = React.forwardRef(function Masonry(inProps, ref) {
 
   const classes = useUtilityClasses(ownerState);
 
-  const handleResize = (masonryChildren) => {
-    if (!masonryRef.current || !masonryChildren || masonryChildren.length === 0) {
-      return;
-    }
+  useEnhancedEffect(() => {
+    const handleResize = (masonryChildren) => {
```

Comment:
Is it necessary to move `handleResize` into the effect? I would rather keep it out, if possible.

Reply:
I had to move it inside because of a eslint rule. I'll see if I can get it working without moving it in and not having to wrap `handleResize` in a `useCallback`.

Reply:
Should be good now, let me know what you think.